zipline/algorithm.py

```diff
@@ -611,16 +612,28 @@ def run(self, data=None, overwrite_sim_params=True):
 
             if isinstance(data, pd.Panel):
                 # For compatibility with existing examples allow start/end
-                # to be inferred.
+                # to be inferred. Start and end have to be midnight UTC.
                 if overwrite_sim_params:
-                    self.sim_params.period_start = data.major_axis[0]
-                    self.sim_params.period_end = data.major_axis[-1]
+                    self.sim_params.period_start = normalize_date(
+                        data.major_axis[0].tz_convert('UTC')
```

Comment:
if it is naive, I think we can assume it is utc by tz_localizing to utc iff it is tz-naive